engine/src/engine.rs

```diff
@@ -607,11 +607,16 @@ impl Engine {
         Ok(SubmitResult::new(status, used_gas, logs))
     }
 
-    pub fn call_with_args(&mut self, args: FunctionCallArgs) -> EngineResult<SubmitResult> {
-        let origin = self.origin();
-        let contract = Address(args.contract);
-        let value = Wei::zero();
-        self.call(origin, contract, value, args.input, u64::MAX, Vec::new())
+    /// Call the EVM contract with arguments (attached ETH balance, input, gas limit, etc.)
+    pub fn call_with_args(&mut self, args: impl Into<CallArgsType>) -> EngineResult<SubmitResult> {
```

Comment:
I think we should simply change the signature here to only accept `CallArgsType`. The conversion from the legacy type can be done at the call-site if needed.